src/spring.js

```diff
@@ -0,0 +1,54 @@
+import { createEndValueType, FRAME_RATE, noSpeed, updateCurrV, updateCurrVals } from './utils';
+import Motion from './motion';
+import { PropTypes } from 'react';
+
+export default class Spring extends Motion {
```

Comment:
I'm ok with have both Spring and TransitionSpring be as they were instead of extending a new class. However, it's almost 40 lines of code that they share and, unless they are to be completely different, that's a great deal of overhead. Motion works more like an abstract class in a way. If that pattern is kept, perhaps the [invariant check](https://github.com/dariocravero/react-motion/commit/f68cca96e3322b3d7b55110eaf290e76d277af77#diff-c333f12887279f59e771971f7607eb37R9) can be replaced by something like `raf() { throw Error('implement it!') }`.